openvino_xai/api/api.py

```diff
@@ -50,7 +50,6 @@ def insert_xai(
         explain_method=explain_method,
         target_layer=target_layer,
         embed_scaling=embed_scaling,
-        prepare_model=False,
```

Comment:
Please elaborate, why `prepare_model=False`?

Reply:
I removed this line, and then it's `prepare_model=True` by default. It's for auto detect `Method`.
The "CNNs->ViTs" fallback mechanism is implemented in the `WhiteBoxMethodFactory.create_method()` and it works only with `prepared_model=True` because we try to insert XAI head while model creation in `Method.__init__()` by calling `Method.prepare_model()`. Otherwise, users should always specify the `Method` for `insert_xai()` API.

Reply:
But if you see the below line (number 57), there is one more prepare the model (without loading). 
In this case we do prepare model twice to be able to support "CNNs->ViTs" fallback mechanism?

Reply:
I tried to avoid duplicated insertion and compile for that reason.
![image](https://github.com/user-attachments/assets/3259ba2e-d787-4659-9a8b-ad100bc4c6ba)
Once XAI-inserted (self._model is not None) or compiled (self._model_compiled is not None),
prepare_model() would just return the previously processed model.